group/variables.tf

```diff
@@ -1,74 +1,133 @@
   description = "ID of the target VPC."
 }
 
-variable "region" {
+## LC parameters
+variable "ami" {
   type        = "string"
-  description = "AWS region to be utilized."
+  description = "Amazon Machine Image (AMI) to associate with the launch configuration."
 }
 
-variable "subnets" {
+variable "associate_public_ip_address" {
   type        = "string"
-  description = "List of VPC subnets to associate with the auto scaling group."
+  description = "Flag for associating public IP addresses with instances managed by the auto scaling group."
+  default     = ""
 }
 
-## LC parameters
-variable "ami" {
+variable "ebs_optimized" {
   type        = "string"
-  description = "Amazon Machine Image (AMI) to associate with the launch configuration."
+  description = "Flag to enable EBS optimization."
+  default     = "false"
 }
 
-variable "instance_type" {
+variable "ebs_vol_del_on_term" {
   type        = "string"
-  description = "EC2 instance type to associate with the launch configuration."
+  description = "Whether the volume should be destroyed on instance termination."
+  default     = "true"
 }
 
-variable "instance_profile" {
+variable "ebs_vol_device_name" {
   type        = "string"
-  description = "IAM instance profile to associate with the launch configuration."
+  description = "The name of the device to mount."
+  default     = ""
 }
 
-variable "key_name" {
+variable "ebs_vol_encrypted" {
   type        = "string"
-  description = "SSH key pair to associate with the launch configuration."
+  description = "Whether the volume should be encrypted or not. Do not use this option if you are using 'snapshot_id' as the encrypted flag will be determined by the snapshot."
+  default     = ""
 }
 
-variable "associate_public_ip_address" {
+/*
+http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSVolumeTypes.html
+For the best per-I/O latency experience, we recommend that you provision an IOPS-to-GiB ratio greater than 2:1. For example, a 2,000 IOPS volume should be smaller than 1,000 GiB.
+*/
+variable "ebs_vol_iops" {
   type        = "string"
-  description = "Flag for associating public IP addresses with instances managed by the auto scaling group."
-  default     = false
+  description = "The amount of provisioned IOPS"
+  default     = "2000"
 }
 
-variable "user_data" {
+variable "ebs_vol_size" {
   type        = "string"
-  description = "Instance initialization data to associate with the launch configuration."
+  description = "The size of the volume in gigabytes."
+  default     = ""
+}
+
+variable "ebs_vol_snapshot_id" {
+  type        = "string"
+  description = "The Snapshot ID to mount."
+  default     = ""
+}
+
+variable "ebs_vol_type" {
+  type        = "string"
+  description = "The type of volume. Valid values are 'standard', 'gp2' and 'io1'."
+  default     = "gp2"
 }
 
 variable "enable_monitoring" {
   type        = "string"
   description = "Flag to enable detailed monitoring."
-  default     = true
+  default     = ""
```

Comment:
Are `bool: false` and `string: ""` `int: 0`  interchangeable when specifying a flag? It wasn't clear to me in the documentation.